tests/lib/prepare-restore.sh

```diff
@@ -654,6 +670,19 @@ restore_suite_each() {
     # restore test directory saved during prepare
     tests.backup restore
 
+    # Save all the installed packages and remove the new packages installed 
+    if os.query is-classic; then
+        distro_get_installed_packages > installed-final.pkgs
+        diff -u installed-initial.pkgs installed-final.pkgs | grep -E "^\+" | tail -n+2 | cut -c 2- > installed-new.pkgs
+
+        # shellcheck disable=SC2002
+        packages="$(cat test | tr "\n" " ")"
+        if [ -n "$packages" ]; then
+            # shellcheck disable=SC2086
+            distro_purge_package $packages
```

Comment:
should we also mark the test as failed, if it did not properly remove all the packages it installed?

Reply:
well, in case it can't remove it should fail and the test will remain as failed